File: sudoq-app/sudoqmodel/src/main/java/de/sudoq/model/solverGenerator/solution/XWingDerivation.java
package de.sudoq.model.solverGenerator.solution;

import java.util.BitSet;
import java.util.List;
import java.util.Stack;

import de.sudoq.model.solvingAssistant.HintTypes;
import de.sudoq.model.sudoku.Constraint;

/**
 * Created by timo on 04.10.16.
 */
public class XWingDerivation extends SolveDerivation {


    private List<Constraint> lockedConstraints;
    private List<Constraint> reducibleConstraints;
    private int note;

    public XWingDerivation() {
        super(HintTypes.XWing);
        lockedConstraints    = new Stack<>();
        reducibleConstraints = new Stack<>();
    }

    public void setLockedConstraints(Constraint c1, Constraint c2){
        lockedConstraints.add(c1);
        lockedConstraints.add(c2);
    }

    public void setReducibleConstraints(Constraint c1, Constraint c2){
        reducibleConstraints.add(c1);
        reducibleConstraints.add(c2);
    }

    public List<Constraint> getReducibleConstraints() {
        return reducibleConstraints;
    }

    public List<Constraint> getLockedConstraints() {
        return lockedConstraints;
    }

    public void setNote(int i){
        note=i;
    }

    public int getNote() {
        return note;
    }




}

File: sudoq-app/sudoqmodel/src/main/kotlin/de/sudoq/model/solverGenerator/solution/XWingDerivation.kt
package de.sudoq.model.solverGenerator.solution

import de.sudoq.model.solvingAssistant.HintTypes
import de.sudoq.model.sudoku.Constraint
import java.util.*

class XWingDerivation : SolveDerivation(HintTypes.XWing) {

    private val lockedConstraints: MutableList<Constraint> = Stack()
    private val reducibleConstraints: MutableList<Constraint> = Stack()

    var note = 0

    fun setLockedConstraints(c1: Constraint, c2: Constraint) {
        lockedConstraints.add(c1)
        lockedConstraints.add(c2)
    }

    fun setReducibleConstraints(c1: Constraint, c2: Constraint) {
        reducibleConstraints.add(c1)
        reducibleConstraints.add(c2)
    }

    fun getReducibleConstraints(): List<Constraint> {
        return reducibleConstraints
    }

    fun getLockedConstraints(): List<Constraint> {
        return lockedConstraints
    }

    init {
        setDescription("XWing")
    }
}